### ISSUE TITLE
[Feature Request] AWS IoT

### ISSUE THREAD

**Issue by DarkFox** [OttoWinter/esphomelib#191]
Amazon Web Services IoT offers MQTT over Websocket and there are existing implementations for Arduino/ESP8266.

**Comment by OttoWinter** [comment]
Yes in theory that would be nice, though the MQTT client library used by esphomelib only supports communication over a direct TCP socket connection. Supporting websockets probably means re-writing large parts of that library which I do not intend to do in the near future.

**Comment by DarkFox** [comment]
Yeah, I see it's quite tightly coupled. I might give it a go some time though. :)
